### PR TITLE
Add resnet50 example used with onnxruntime inferencing

### PR DESCRIPTION
**Pull request by TedThemistokleous**
skip ci

Useful for sanity checking if we can perform a resnet50 through MIGraphX Execution Provider thats run through onnxruntime.

Run preq_steps.sh to build and install onnxruntime and pytorch for rocm after someone has installed MIGraphX or via apt-get

User can just use this to sanity check their end to end run by first running prereq_steps.sh and then resnet50.py

The code for this example was taken from onnxruntime's documentation on how to download a model and run an inference using tags. Leverages the use of torchvision's resnet50 implementation. 

This was tested using DLM as this pipe will be used to sanity test future cards as an alternative to using our migraphx-driver for resnet50. 

Bonus, there's a cute test image of my bird.

### CODE REVIEW

**Comment by pfultz2** [comment]
Please add a README.md to describe how one would run the example.

**Comment by TedThemistokleous** [comment]
> Please add a README.md to describe how one would run the example.

Done

File: examples/onnxruntime/resnet50/README.txt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
```
Comment (pfultz2) [inline examples/onnxruntime/resnet50/README.txt:23]: I dont think we need a license for the readme. I think if you use .md instead .txt it should not require the license in CI.

File: examples/onnxruntime/resnet50/README.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+
+ To run the example please do the following:
+
+- Build and install MIGraphX as a package using "make package && dpkg -i *.deb" in the build folder via the Docker build method
+- Move to this folder examples/onnxruntime/resnet50
+- Run the prequisite install script ./prereq_steps.sh - this installs pytorch
+- The prequisite install script will also build and install onnxruntime on your system. Ensure this with pip list | grep onnxruntime
+- run the resnet50 inference via python resnet50.py
```
Comment (pfultz2) [inline examples/onnxruntime/resnet50/README.txt:32]: Can you write out the commands in a markdown code block?

Comment (TedThemistokleous) [reply examples/onnxruntime/resnet50/README.txt]: Done

File: examples/onnxruntime/resnet50/README.txt (hunk 1 above)
Comment (TedThemistokleous) [reply examples/onnxruntime/resnet50/README.txt]: Removed

File: examples/onnxruntime/resnet50/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## Description
+
+This example demonstrates how to perform an MIGraphX Python API inference through onnxruntime. The model used here is from Torchvision's pretrained resnet50 model
+
+## Content
+- [Basic Setup](#Basic-Setup)
+- [**Running this Example**](#Running-this-Example)
+
+## Basic Setup
+Before running inference we must first install MIGraphX via the Docker method as it also downloads onnxruntime into the dockerfile created.
```
[the remark this answers is not in the document]
Comment (TedThemistokleous) [reply examples/onnxruntime/resnet50/README.md]: Added steps and info on preq steps before running this example

**Comment by TedThemistokleous** [comment]
Looks like there isn't a cockatiel bird label so this test without the cat, is now assuming the picture of my bird is an African Grey..

File: examples/onnxruntime/resnet50/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+Before running inference we must first install MIGraphX via the Docker method as it also downloads onnxruntime into the dockerfile created. 
+
+Starting from project root:
+```
+$ cd AMDMIGraphX
+$ docker build -t migraphx .
+$ docker --device='/dev/kfd' --device='/dev/dri' <your docker settings and mounts> --group-add video -it migraphx
+```
+
+The dockerfile will install the latest supported version of ROCm with all the depandacnies needed for MIGraphX
```
Comment (umangyadav) [inline examples/onnxruntime/resnet50/README.md:22]:
```suggestion
The dockerfile will install the latest supported version of ROCm with all the dependencies needed for MIGraphX
```

**Review by umangyadav — APPROVED** [review]

**Review by causten — CHANGES_REQUESTED** [review]
Can you reduce the image sizes of your jpgs.  Each of the jpgs are ~2.2 MB.  Since resnet50 uses a 224x224 image there really is no need to have 13 MB of space for 6 pictures when a user clones the repo

**Comment by TedThemistokleous** [comment]
> Can you reduce the image sizes of your jpgs. Each of the jpgs are ~2.2 MB. Since resnet50 uses a 224x224 image there really is no need to have 13 MB of space for 6 pictures when a user clones the repo

Done. Re-scaled and updated metadata so the images I took are tagged to be public domain via exif data.

**Comment by TedThemistokleous** [comment]
Update 1/15/2024

Added additional input args to the python script to specify things like batch size, verbosity, as well as different image dirs and fp16 while ensuring we do proper IO binding with Onnxruntime. 

I'll be using this to run DeepLearningModels through Onnxruntime and resnet50 to ensure we can get an inference. Let me know if there's anything I need to fix add. 

Yes I need to adjust QPS. This should pull down and run pytorchs resnet50 v1.5 with the latest weights before preprocessing data

**Review by causten — APPROVED** [review]

**Comment by causten** [comment]
Looks like a format problem, after that I will merge

**Comment by TedThemistokleous** [comment]
> Looks like a format problem, after that I will merge

Hopefully latest works. VS code wasn't running my githooks with many repos open at once.

**Pull request merged**